### DISCUSSION TITLE
Why can't I access password in user? (ORM) (Postgres-Supabase)

### DISCUSSION THREAD

**Discussion by Gaurav-Narayan-Varma** [prisma/prisma#24397] [Q&A]
### Bug description

Using the following query:
```
    const user = await prisma.user.findFirst({ where: { email: email } });

    console.log("user:", user);
```
Produces the following in the terminal:
```
user: { id: 5, email: 'user@nextmail.com', name: 'asdfvc', example: 'asdf' }
```
Even though my users table contains a password field and an entered password in my supabase and it is in sync with my schema. 

Interestingly enough, when I drop the example column and then push, and then run this:

```
    const user = await prisma.user.findFirst({
      where: { email: email },
      select: {
        id: true,
        email: true,
        password: true,
      },
    });
```

I get the following:

```
Failed to fetch user: PrismaClientValidationError: 
Invalid `prisma.user.findFirst()` invocation:

{
  where: {
    email: "user@nextmail.com"
  },
  select: {
    id: true,
    email: true,
    password: true,
    ~~~~~~~~
?   name?: true,
?   example?: true
  }
}
```
Which means it's still expecting the example field. Why could that be?

Any thoughts on why the prisma query won't select password?

### Expected behavior

The query should return a user with the password field

### Prisma information

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Service {
  id               Int      @id @default(autoincrement())
  html             String
  editor_state     String
  created_at       DateTime @default(now())
  updated_at       DateTime @updatedAt
  name             String
  slug             String
  card_title       String
  card_description String
}

model User {
  id         Int      @id @default(autoincrement())
  email      String   @unique
  password String @default("$2a$10$cQEFiUaWpFvVVO5AcV3QN.7ohJXxxOPouTxORiZP.mUhupwEpDnUy")
  created_at DateTime @default(now())
  updated_at DateTime @default(now()) @updatedAt
  name String
}
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: MacOS 14.4.1
- Database: PostgresSQL
- Node.js version: 20.12.2
- prisma/client:  ^5.14.0

### Prisma Version

```
5.14.0
```

**Comment by janpio** [comment]
It looks like the Prisma Client you are using in your app is not up-to-date with your Prisma schema - first it is missing the `password`, and then later still having the `example`.
Did you run `prisma generate`?

**Reply by Gaurav-Narayan-Varma** [reply]
Yes I ran prisma generate. I actually had to close and reopen my VS Code to resolve this issue- this post provided the answer: https://stackoverflow.com/questions/65663292/prisma-schema-not-updating-properly-after-adding-new-fields.

My guess is  there was a bug with TS not tracking the prisma changes so I had to restart it?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
